src/OpenLoco/GameCommands/RenameCompanyOwner.cpp

```diff
@@ -0,0 +1,121 @@
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Miscellaneous);
+
+        // Keep track of the company id over several calls.
+        static CompanyId _companyId{};
+        if (index == 1)
+            _companyId = companyId;
+
+        static uint32_t renameBuffer[9];
+
+        // Fill buffer over calls into the renameBuffer
+        if ((flags & GameCommands::Flags::apply) != 0)
+        {
+            static const std::array<int, 3> transformTable = { 2, 0, 1 };
+            int arrayIndex = transformTable.at(index);
+            renameBuffer[arrayIndex * 3] = buffer0;
+            renameBuffer[arrayIndex * 3 + 1] = buffer1;
+            renameBuffer[arrayIndex * 3 + 2] = buffer2;
+        }
+
+        // Applying the buffer?
+        if (index != 0)
+            return 0;
+
+        char renameStringBuffer[37] = "";
+        memcpy(renameStringBuffer, renameBuffer, sizeof(renameBuffer));
+        renameStringBuffer[36] = '\0';
+
+        // Ensure the new name isn't empty.
+        if (strlen(renameStringBuffer) == 0)
+            return 0;
+
+        // Retrieve the current name for this company.
+        char currentOwnerName[256] = "";
+        auto company = CompanyManager::get(_companyId);
+        StringManager::formatString(currentOwnerName, company->ownerName);
+
+        // Verify the new name actually differs from the old one.
+        if (strcmp(currentOwnerName, renameStringBuffer) == 0)
+            return 0;
+
+        // Verify the name isn't in use by any other company
+        for (auto& rival : CompanyManager::companies())
+        {
+            if (rival.id() == _companyId)
+                continue;
+
+            char rivalOwnerName[256] = "";
+            StringManager::formatString(rivalOwnerName, rival.ownerName);
+
+            if (strcmp(rivalOwnerName, renameStringBuffer) == 0)
+            {
+                GameCommands::setErrorText(StringIds::chosen_name_in_use);
+                return GameCommands::FAILURE;
+            }
+        }
+
+        // Allocate a string id for the new name.
+        string_id allocatedStringId = StringManager::userStringAllocate(renameStringBuffer, 0);
+        if (allocatedStringId == StringIds::empty)
+            return GameCommands::FAILURE;
+
+        // Bailing out early?
+        if ((flags & GameCommands::Flags::apply) == 0)
+        {
+            StringManager::emptyUserString(allocatedStringId);
+            return 0;
+        }
+
+        // Apply the new owner name to the company.
+        string_id oldStringId = company->ownerName;
+        company->ownerName = allocatedStringId;
+        StringManager::emptyUserString(oldStringId);
+        Gfx::invalidateScreen();
+        sub_434BA1();
+        return 0;
+    }
+
+    void changeCompanyOwnerName(registers& regs)
+    {
+        regs.ebx = changeCompanyOwnerName(regs.bl, CompanyId(regs.cl), regs.ax, regs.edx, regs.ebp, regs.edi);
+    }
+
+    // 0x00434BA1
+    void sub_434BA1()
```

Comment:
This function is doing something like updating the company competitor face not sure exactly what is happening its a bit weird.

Reply:
Yeah, I thought it was a bit odd, too. It's only used by this game command, too. Decided to leave it for another PR, though.